module.c

```diff
@@ -267,26 +266,19 @@ void mod_update_list()
   module_t *ml, **ml1;
   struct dirent *de;
   DIR *d;
-  char buf[32];
-  int i, found;
   size_t ext_pos;
 
   for(ml1 = &config.module.list; *ml1; ml1 = &(*ml1)->next) (*ml1)->exists = 0;
 
   if(!(d = opendir(config.module.dir))) return;
 
   while((de = readdir(d))) {
-    i = strlen(de->d_name);
-    if(
-      i < (int) sizeof buf &&
-      mod_has_module_ext(de->d_name, i, &ext_pos)
-    ) {
-      strcpy(buf, de->d_name);
-      buf[ext_pos] = 0;
+    if(mod_has_module_ext(de->d_name, strlen(de->d_name), &ext_pos)) {
+      int found = 0;
```

Comment:
This is not correct as it edits out that `buf[ext_pos] = 0`. This was actually the reason for that duplicate buffer.